context-monitoring/src/main/java/de/atb/context/monitoring/monitors/file/AbstractFileSystemMonitor.java

```diff
@@ -0,0 +1,302 @@
+                            this.logger.debug(cwse.getMessage(), cwse);
+                            break;
+                        }
+                    }
+                } catch (IOException e) {
+                    this.logger.error(e.getMessage(), e);
+                }
+            }
+        }, "File watch service Thread");
+        this.watchDaemon.start();
+    }
+
+    protected final void handleWatchEvents(final List<WatchEvent<?>> events,
+                                           final Long time) {
+        String watchedPath = String.valueOf(Paths.get(this.pathToMonitor
+            .getAbsolutePath()));
+        String from = null;
+        for (WatchEvent<?> e : events) {
+            Path context = (Path) e.context();
+            String file = watchedPath + java.io.File.separator
+                + context.toString();
+            InterpreterConfiguration setting = this.interpreter
+                .getConfiguration(file);
+            if (e.kind() == StandardWatchEventKinds.ENTRY_CREATE) {
+                fileCreated(file, time, setting);
+            } else if (e.kind() == StandardWatchEventKinds.ENTRY_MODIFY) {
+                fileModified(file, time, setting);
+            } else if (e.kind() == StandardWatchEventKinds.ENTRY_DELETE) {
+                fileDeleted(file, time, setting);
+            } else {
+                this.logger.debug("Event " + e.kind() + " will be ignored");
+            }
+        }
+    }
+
+    protected abstract void iterateFiles(final File directory);
+
+    protected abstract void handleFile(final String fileName, final Long time,
+                                    final InterpreterConfiguration setting);
+
+    protected abstract IndexingParser<P> getParser(final InterpreterConfiguration setting);
+
+    protected final void fileExisting(final String file, final Long time,
+                                      final InterpreterConfiguration setting) {
+        if (this.logger.isTraceEnabled()) {
+            this.logger.trace(file
+                + " already existed at "
+                + this.getDefaultDateFormat().format(
+                new Date(time)));
+        }
+        if (setting != null) {
+            this.filesToDates.put(file, time);
+        }
+        handleFile(file, time, setting);
+    }
+
+    protected final void fileCreated(final String file, final Long time,
+                                     final InterpreterConfiguration setting) {
+        if (this.logger.isTraceEnabled()) {
+            this.logger.trace(file
+                + " created at "
+                + this.getDefaultDateFormat().format(
+                new Date(time)));
+        }
+        if (setting != null) {
+            this.filesToDates.put(file, time);
+        }
+        handleFile(file, time, setting);
+    }
+
+    protected final void fileDeleted(final String file, final Long time,
+                                     final InterpreterConfiguration setting) {
```

Comment:
`setting` is never used and can be removed

Reply:
[x] done